apps/aecore/src/aec_tx_sign.erl

```diff
@@ -1,23 +1,100 @@
 -module(aec_tx_sign).
 
+%% @doc Implements a data structure for cryptographically signed transactions.
+%% This is the envelope around transactions to make them cryptographically safe. 
+%% The transactions normally also have keys of the "signers" in the transaction,
+%% which are extracted using the signers/1 function in the respective transaction
+%% handler.
+%%
+%% The purpose of this module is to provide an API for cryptograpically signed 
+%% transactions and hide all implementation details. Therefore, the record 
+%% #signed_tx{} should be kept private and considered an abstract type.
+%%
+%% A transaction can be signed by one or several signers. Each transaction can
+%% determine its own signers by the transaction callback 'signers'. Since we do not
+%% want to depend upon transaction types in this module, the user of
+%% {@module} should first obtain the signers of the transaction and then call this
+%% {@link sign/2} with these signers. There is a {@link sign/3} function that can sign
+%% with respect to a certain block height. This is handy whenever the governance
+%% variables on what crypto to use would change.
+
 %% API
--export([data/1,
+-export([sign/2,
+         sign/3, 
+         data/1,
          signatures/1,
-         verify/1]).
--export([serialize/1,
+         verify/2]).
+-export([verify/1,
+         serialize/1,
          serialize_to_binary/1,
          deserialize/1,
          deserialize_from_binary/1]).
 
 -include("common.hrl").
--include("txs.hrl").
 
+-record(signed_tx, {
+          data                       :: term(),
```

Comment:
I think we can find out what this is. Probably a couple of transaction types.

Reply:
Actually, signed_tx should be a parameterized type signed_tx(tx_coinbase()) for example. But a bit too much to fix that in this PT